test/server/config_validation/xds_verifier.cc

```diff
@@ -139,24 +159,30 @@ void XdsVerifier::listenerAdded(const envoy::config::listener::v3::Listener& lis
  */
 void XdsVerifier::listenerRemoved(const std::string& name) {
   bool found = false;
-  for (unsigned long i = 0; i < listeners_.size(); ++i) {
-    auto& rep = listeners_[i];
+
+  for (auto it = listeners_.begin(); it != listeners_.end();) {
+    auto& rep = *it;
     if (rep.listener.name() != name) {
+      ++it;
```

Comment:
consider inverting the != name condition to == name, so that you can combine all the iterator advances at the end